e2e/tests-dfx/call.bash

```diff
@@ -26,9 +26,9 @@ teardown() {
   CANISTER_ID=$(dfx canister id hello_backend)
   rm .dfx/local/canister_ids.json
 
-  # if no candid file known, then no field names
+  # given a canister id, fetch the did file form metadata
   assert_command dfx canister call "$CANISTER_ID" make_struct '("A", "B")'
-  assert_eq '(record { 99 = "A"; 100 = "B" })'
+  assert_eq '(record { c = "A"; d = "B" })'
```

Comment:
Make sense. It's easier the other way around: we test fetching the metadata given a canister id in this test; in the next test, we test the `--candid` override by using an empty did file, so that we don't get field names.

Reply:
I must be missing something.  After these change, where is a test that demonstrates `--candid <valid candid path>` can provide the field names?  With current changes all this demonstrates is that passing `--candid empty.did` can hide them.

Reply:
I think we want to test that `--candid` is working as expect, i.e., providing a `--candid` flag overrides what's in the metadata. Whether we get field names or not is a side effect of which did file we use, but not what we really want to test.

Reply:
If I altered the `canister call` such that the `--candid` argument unconditionally discarded all candid info, would this test pass?

Reply:
Ok, let's pass in `record { c : Text }` as the return type in `--candid`. Then the test would show that only field c is displayed.

Reply:
This test demonstrates that from a state where `canister call` does not display field names (an indicator that candid data is not available), passing in a valid `--candid` will display field names (an indicator that candid data is available).

The test as altered no longer does that.

Please revert this test, update it as necessary to test what it was testing before with the new metadata fetching code (meaning: it calls a canister from which it can't fetch candid metadata).  You can copy the and make a new test to verify the new `given a canister id, fetch the did file form metadata` behavior.

Reply:
Done. It's not easy to do this in Motoko, due to the tmp_hack fallback. I have to manually build a wasm.